pages/scatterplot.vue

```diff
@@ -0,0 +1,337 @@
+    .text(scatterplotY.value)
+
+  // create the X and Y axis groups
+  const xAxisGroup = svg.append("g")
+    .classed("x-axis", true)
+    .call(xAxis)
+
+  const yAxisGroup = svg.append("g")
+    .classed("y-axis", true)
+    .call(yAxis)
+
+  // Use a group with a transform to contain both the points and the labels
+  const pointsAndLabels = svg.append("g")
+    .classed("points-and-labels", true)
+    .attr("fill", "steelblue")
+    .attr("font-family", "sans-serif")
+    .attr("font-size", 10)
+    .selectAll("g")
+    .data(dataByGrainType.value.get(activeGrainType.value))
+    .join("g")
+    .attr("transform", d => `translate(${x(d[scatterplotX.value])},${y(d[scatterplotY.value])})`)
+
+  // create the scatterplot points
+  const points = pointsAndLabels.append("circle")
+    .attr("r", 3)
+
+  // Label the scatterplot points with the oblast name
+  const labels = pointsAndLabels.append("text")
+    .classed('oblast-label', true)
+    .attr("dy", "0.35em")
+    .attr("dx", "0.35em")
+    .text(d => d.oblastNameUkrainian)
+
+  // select and update all of the point label text
+  labels
+    .attr("fill", "black")
+    .attr("font-weight", "normal")
+    // .filter(d => d.oblastNameUkrainian === activeOblast.value)
+    .attr("fill", "red")
+    .attr("font-weight", "bold")
+    .text(d => d.oblastNameUkrainian)
+
+}
+
+// Select and update the axes and points so that they transition
+// select by the class name
+function updateScatterplot() {
+  const svg = d3.select(scatterplotSvg.value)
+
+  // Get the extent of the scatterplot X and Y in the data
+  const xExtent = d3.extent(dataByGrainType.value.get(activeGrainType.value), (d) => d[scatterplotX.value])
+
+  const yExtent = d3.extent(dataByGrainType.value.get(activeGrainType.value), (d) => d[scatterplotY.value])
+
+  // create the Y and X axes for the scatterplot
+  const x = d3.scaleLinear()
+    .domain(xExtent)
+    .range([margin.left, width - margin.right])
+
+  const y = d3.scaleLinear()
+    .domain(yExtent)
+    .range([height - margin.bottom, margin.top])
+
+  // const pointsAndLabels = svg.append("g")
+  //   .classed("points-and-labels", true)
+  //   .attr("fill", "steelblue")
+  //   .attr("font-family", "sans-serif")
+  //   .attr("font-size", 10)
+  //   .selectAll("g")
+  //   .data(dataByGrainType.value.get(activeGrainType.value))
+  //   .join("g")
+  //   .attr("transform", d => `translate(${x(d[scatterplotX.value])},${y(d[scatterplotY.value])})`)
+
+  // instead of appending select and update with a transition
+  const pointsAndLabels = svg.select(".points-and-labels")
+    .selectAll("g")
+    .data(dataByGrainType.value.get(activeGrainType.value))
+    .join("g")
+    .attr("transform", d => `translate(${x(d[scatterplotX.value])},${y(d[scatterplotY.value])})`)
+    .transition()
+    .duration(1000)
+
+  // update axes and axes labels
+  const xAxis = d3.axisBottom(x)
+  const yAxis = d3.axisLeft(y)
+
+  // clear old axes
+  svg.select(".x-axis").remove()
```

Comment:
We could get rid of these `.remove()` lines using the following:

```js
  const xAxisGroup = svg.selectAll("g.x-axis").data([null]).join("g")
    .classed("x-axis", true)
    .attr("transform", `translate(0,${height - margin.bottom})`)
    .call(xAxis)
```

Reply:
I find this kind of confusing, whereas .remove is very clear